pandas/_libs/lib.pyx

```diff
@@ -137,7 +137,7 @@ except ImportError:
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def memory_usage_of_objects(arr: object[:]) -> int64_t:
+def memory_usage_of_objects(arr: ndarray[object]) -> int64_t:
```

Comment:
This produces:

```
  pandas/_libs/lib.cpython-310-x86_64-linux-gnu.so.p/pandas/_libs/lib.pyx.c:90806:12: warning: '__pyx_memview_set_object' defined but not used [-Wunused-function]
  90806 | static int __pyx_memview_set_object(const char *itemp, PyObject *obj) {
        |            ^~~~~~~~~~~~~~~~~~~~~~~~
  pandas/_libs/lib.cpython-310-x86_64-linux-gnu.so.p/pandas/_libs/lib.pyx.c:90801:20: warning: '__pyx_memview_get_object' defined but not used [-Wunused-function]
  90801 |   static PyObject *__pyx_memview_get_object(const char *itemp) {
        |                    ^~~~~~~~~~~~~~~~~~~~~~~~
```

although I am not sure why it wouldn't let me specify the `ndim` here.